src/main/java/core/basesyntax/strategy/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final Storage storage;
+
+    public PurchaseOperationHandler(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        int currentQuantity = storage.getQuantity(transaction.getFruit());
+        if (currentQuantity >= transaction.getQuantity()) {
+            storage.updateInventory(transaction.getFruit(), -transaction.getQuantity());
+        } else {
+            throw new RuntimeException("Not enough fruits for purchase: "
+                    + transaction.getFruit());
+        }
```

Comment:
avoid if-else construction 
```suggestion
        if (currentQuantity < transaction.getQuantity()) {
            throw new RuntimeException("Not enough fruits for purchase: "
                    + transaction.getFruit());
        }
        storage.updateInventory(transaction.getFruit(), -transaction.getQuantity());
```

Reply:
Ok, thank you!